.grit/patterns/js/calling-set-state-on-current-state.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Calling setState on the current state is always a no-op. so change the state like $Y(!$X) instead.
+---
+
+Calling setState on the current state is always a no-op. Did you mean to change the state like $Y(!$X) instead?
+
+tags: #fix
+
+```grit
+engine marzano(0.1)
+language js
+
+`const $name = ($props) => {$body}` where {
```

Comment:
This is a bit too prescriptive on the format of function components. I'd suggest:

- Create a shared pattern `react_functional_component($props, $body)` in react.grit that is an `or` of the different ways you would define a component.
- Replace that with this.

Reply:
Hi @morgante, updated the suggested changes